logging/include/cocaine/idl/logging.hpp

```diff
@@ -0,0 +1,228 @@
+/*
+    Copyright (c) 2011-2015 Anton Matveenko <antmat@me.com>
+    Copyright (c) 2011-2015 Other contributors as noted in the AUTHORS file.
+
+    This file is part of Cocaine.
+
+    Cocaine is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    Cocaine is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "cocaine/logging/filter.hpp"
+#include "cocaine/logging/attribute.hpp"
+
+#include <cocaine/rpc/protocol.hpp>
+
+#include <boost/mpl/list.hpp>
+
+#include <blackhole/attribute.hpp>
+
+namespace cocaine { namespace io {
+
+struct base_log_tag;
+struct named_log_tag;
+
+// Logging service interface
+
+struct base_log {
+    struct get {
+        typedef base_log_tag tag;
+
+        static const char* alias() {
+            return "get";
+        }
+
+        typedef boost::mpl::list<
+            // Name used for log routing and filtering.
+            std::string
+        >::type argument_type;
+
+        typedef named_log_tag dispatch_type;
+    };
+
+    struct list_loggers {
+        typedef base_log_tag tag;
+
+        static const char* alias() {
+            return "list_loggers";
+        }
+
+        typedef option_of<
+            std::vector<std::string>
+        >::tag upstream_type;
+    };
+
+    struct set_filter {
+        typedef base_log_tag tag;
+
+        static const char* alias() {
+            return "set_filter";
+        }
+
+        typedef boost::mpl::list<
+            // Logger name
+            std::string,
+            // Filter itself
+            logging::filter_t,
+            // Time to store filter
+            uint64_t
```

Comment:
Seconds of course: `typedef std::uint64_t seconds_t;`. Noone needs finer resolution for filters TTL. If they are - fuck them.